imohash/imohash.py

```diff
@@ -12,18 +12,21 @@
 SAMPLE_THRESHOLD = 128 * 1024
 SAMPLE_SIZE = 16 * 1024
 
-def hashfile(filename, sample_threshhold=SAMPLE_THRESHOLD, sample_size=SAMPLE_SIZE, hexdigest=False):
-    size = os.path.getsize(filename)
-
-    with open(filename, 'rb') as f:
-        if size < sample_threshhold or sample_size < 1:
-            data = f.read()
-        else:
-            data = f.read(sample_size)
-            f.seek(size//2)
-            data += f.read(sample_size)
-            f.seek(-sample_size, os.SEEK_END)
-            data += f.read(sample_size)
+#Hashes an opened file object. Compatible with paramimo SFTPFile and regular files.
+def hashfileobject(f, sample_threshhold=SAMPLE_THRESHOLD, sample_size=SAMPLE_SIZE, hexdigest=False):
+    #get file size from file object
+    size = os.fstat(f.fileno()).st_size
+
+    if size < sample_threshhold or sample_size < 1:
+        data = f.read()
+    else:
+        data = f.read(sample_size)
+        f.seek(size//2)
+        data += f.read(sample_size)
+        f.seek(-sample_size, os.SEEK_END)
+        data += f.read(sample_size)
+
+    f.close()
```

Comment:
Since hashfileobject didn't open `f`, I don't think it should close it. I'd remove this and put something like`with open(filename, 'rb') as f:` at line 39